### DISCUSSION TITLE
Time convergence issue while line reconnection

### DISCUSSION THREAD

**Discussion by PandaRP01** [CURENT/andes#481] [Q&A]
Hello Dr. Cui

I am facing a time convergence issues while making line connection and isolation. Kindly suggest the solution. The description of the issues are as follows

The issues remains same for both the scenarios.

**Scenario: 1** 
A line is connecting two power system network which have indivisual slack buses. Initially `Line.u=1`, simulation is started, Initialization was successful. TDS `(TDS.config.tf=40)` was running fine.
when line is disconnected (`Line.u=0`), ran the simulation (`TDS.config.tf=60`) and TDS was completed successfully.
 Scenarios:1
 Then I reconnected the Line (`Line.u=1`) and ran the TDS (`TDS.config.tf=80`), `TDS was terminated as Time step reduced to zero. Convergence is not likely.` The error massage was listed below.

 **Scenario: 2** 
 I thought, this error may be due to different voltage and frequecy of the intrconnecting buses. Hence I made the load active power and reactive at one side and tried to bring down  the one of the interconnecting bus volatge and frequency approxmiatly equal to the other interconnecting bus voltage. Then I reconnected the line (`Line.u=1`). However, the error remain same, i.e `TDS was terminated as Time step reduced to zero. Convergence is not likely`. The error massage was listed below.
 
 The error massage are as follows
 
```
 C:\Users\rakeshpa\.andes\pycode\ESST3A.py:21: RuntimeWarning: invalid value encountered in sqrt
  return (-v + vbus, LAW2_y*VB_y*ue - vout, -UEL + UEL0, ue*(-IN*VE + KC*XadIfd), -FEX_y + select([less_equal(IN, 0),less_equal(IN, 
0.433),less_equal(IN, 0.75),less_equal(IN, 1),greater(IN, 1),__trues], [__ones,1 - 0.577*IN,sqrt(0.75 - IN**2),1.732 - 
1.732*IN,__zeros,__zeros], default=nan), FEX_y*VE - VB_x, VBMAX*VB_lim_zu + VB_lim_zi*VB_x - VB_y, KG*vout - VG_x, 
VGMAX*VG_lim_zu + VG_lim_zi*VG_x - VG_y, LAW1_y - VG_y - vrs, -vref + vref0, -LG_y - vi + vref, HLI_zi*vi + HLI_zl*VIMIN + 
HLI_zu*VIMAX - vil, HG_lt_z0*UEL + HG_lt_z1*vil - HG_y, LL_LT1_z1*LL_LT2_z1*(-LL_x + LL_y) + LL_x*TB - LL_y*TB + TC*(HG_y - LL_x), 
ue*(-vf0 + vout), 0, 0, 0, 0, 0, 0, 0,)
```
  
```
  C:\Users\rakeshpa\.andes\pycode\ESST3A.py:33: RuntimeWarning: invalid value encountered in sqrt
  return (-1, 1, -1, LAW2_y*ue, -1, -VE*ue, KC*ue, select([less_equal(IN, 0),less_equal(IN, 0.433),less_equal(IN, 0.75),less_equal(IN, 
1),__trues], [__zeros,-0.577*__ones,-IN/sqrt(0.75 - IN**2),-1.732*__ones,__zeros], default=nan), -1, VE, -1, VB_lim_zi, -1, KG, -1, VG_lim_zi,
 -1, -1, -1, -1, 1, -1, HLI_zi, -1, HG_lt_z0, HG_lt_z1, -1, TC, LL_LT1_z1*LL_LT2_z1 - TB, ue)
```
  
  
```
Time step reduced to zero. Convergence is not likely.
Simulation terminated at t=70.0000 s.
Simulation to t=80.00 sec completed in 15.9048 second
```

Regards
Rakesh Panda

**Comment by jinningwang** [comment]
Line trip is expected to do with ``Toggle``, see [case IEEE 14-bus line trip](https://github.com/CURENT/andes/blob/master/andes/cases/ieee14/ieee14_linetrip.xlsx) and [Example 9](https://docs.andes.app/en/latest/_examples/ex9.html) for how to do the line trip.

Not sure if manually altering ``Line.u`` value will cause unexpected errors, but using ``Toggle`` should give you reasonable results.

**Reply by PandaRP01** [reply]
Hello jinningwang

Thanks for your response. I could use toggle for line trip, however, as I make it flexible the time of operation for line switching, I was doing it through pert file. 

I will try it using toggle, and let you know the response.

Thank you and Regards
Rakesh
